core/src/main/java/com/scalar/db/transaction/consensuscommit/CommitHandlerWithGroupCommit.java

```diff
@@ -57,6 +57,23 @@ public void commit(Snapshot snapshot, boolean readOnly)
     super.commit(snapshot, readOnly);
   }
 
+  @Override
+  boolean canOnePhaseCommit(Snapshot snapshot) throws CommitException {
+    try {
+      return super.canOnePhaseCommit(snapshot);
+    } catch (CommitException e) {
+      cancelGroupCommitIfNeeded(snapshot.getId());
```

Comment:
If `canOnePhaseCommit()` fails, we need to cancel the group commit.

Reply:
Can you explain the reason why we need to cancel group commit in case `canOnePhaseCommit` throws an exception? Since `canOnePhaseCommit` (sounds like) haven't done one-phase commit yet, It looks like there is no need to cancel the group commit.

Reply:
@feeblefakie 

>  if canOnePhaseCommit for a transaction throws an exception, we just give up executing the transaction instead of falling back to executing it in the normal (non-one-phase) way?

Yes, we just give up executing the transaction.

> Giving up is required for correctness? or we just do it for simplicity?

We just do it for simplicity. Since the situation where `canOnePhaseCommit` throws an exception is very rare, I believe the simpler approach (giving up executing the transaction instead of falling back to executing it in the normal way) is better. Thanks.

Reply:
Got it. Thanks!